oauth2client/_transport/urllib3.py

```diff
@@ -163,13 +166,19 @@ def request(http_object, uri, method='GET', body=None, headers=None, **kwargs):
     Args:
         http_object (urllib3.request.RequestMethods): Any instance that
             provides the :class:`RequestMethods` interface.
-        uri (string): The URI to be requested.
-        method (string): The HTTP method to use for the request. Defaults
+        uri (str): The URI to be requested.
+        method (str): The HTTP method to use for the request. Defaults
             to 'GET'.
-        body (string): The payload / body in HTTP request.
-        headers (Mapping): Key-value pairs of request headers.
+        body (bytes): The payload / body in HTTP request.
+        headers (Mapping): Request headers.
     Returns:
         urllib3.response.HTTPResponse: The HTTP response.
     """
+
+    # Urllib3 uses a sentinel default value for timeout, so only set it if
+    # specified.
+    if timeout is not None:
```

Comment:
Do you not want to use a sentinel as well? Like what if a user or internal caller pass an explicit `None`?